pages/blog/20190831-newfeature.md

```diff
@@ -0,0 +1,124 @@
+# Adding a New Feature
+
+31, August 2019 • [irisb1701](https://github.com/irisb1701)
+
+---
+
+As a system's engineer you may need to work with the treehouses cli (command line interface). 
+
+Here are step by step instructions on how to create a new feature for treehouses.
+
+## Determining a New Feature:
+
+Example situation:
+As you're testing the new image for treehouses you notice your raspberry pi is getting hot. You know anything past 85°C may damage your pi so you'd like to monitor its core temperature.
+
+You already know the raspberry pi command `vcgencmd measure_temp` will return its core temperature in celsius.
+
+Looking at the treehouses cli commands, you notice there aren't any commands involving detecting temperature and decide it would be a useful addition.
+
+Now that you have an idea for a new command, you will create a new issue and when approved, can create a new branch and begin working on it.
+
+
+## Step 0 - Vim
+
+Vim is an easy way to test your code. Here are instructions on how to use it.
+
+1. Download Vim [here](https://www.vim.org/download.php) for windows and follow their instructions for set-up or
+    1. type `brew install vim` in your terminal for Mac
+    1. `sudo apt-get install vim` in your terminal for Ubuntu
+1. Create a new file and give it a name`vim mycoolnewfile`
+1. The new file will automatically open, now press i on your keyboard to edit it
+1. Include `#!/bin/bash` at the top
+1. Copy the below script and press `:wq` to save it and exit
+1. ![](../../pages/blog/images/20190831-vim-file.png)
+1. Now enter the command `chmod 755 mycoolnewfile` so you can run it
+1. Enter `./mycoolnewfile`
+1. You should now see what the function returns. This is how you can easily test your functions
+1. ![](../../pages/blog/images/20190831-vim-commands.png)
+
+
+## Step 1: Creating your new Command
+Your function can be as simple as just calling the `vcgencmd measure_temp` command, however this command will return something like "temp=60.00'C".
+You may want to modify it.
+
+1. Your first line will define a function which we will call temperature
+1. Inside the function we will call `vcgencmd measure_temp` by simply writing in the command
+1. ![](../../pages/blog/images/20190831-vgenc.png)
```

Comment:
@Kid243 the floating list number is due to an error in the image that's suppose to show up their. As @phamduchongan93 pointed out, the error is due to the path. I believe it should be `![](images/20190831-vgenc.png)`

Looks great @irisb1701. Excellent job on the tutorial/blog.